lib/active_storage/blob.rb

```diff
@@ -53,6 +53,11 @@ def upload(io)
     self.checksum  = compute_checksum_in_chunks(io)
     self.byte_size = io.size
 
+    if service.nil?
+      raise "No storage service specified for current env (#{Rails.env}). " \
+            "Add config.active_storage.service = :local into your config/environments/#{Rails.env}.rb."
+    end
```

Comment:
If no storage service is specified, we’ll want to throw an error on app boot, not at some unspecified point in the future.